larky/src/main/resources/vendor/Crypto/OpenPGP/cryptography.star

```diff
@@ -0,0 +1,585 @@
+        elif algo == 3:
+            return cipher(algorithms.CAST5, 16, 8)
+        elif algo == 4:
+            return cipher(algorithms.Blowfish, 16, 8)
+        elif algo == 7:
+            return cipher(algorithms.AES, 16, 16)
+        elif algo == 8:
+            return cipher(algorithms.AES, 24, 16)
+        elif algo == 9:
+            return cipher(algorithms.AES, 32, 16)
+
+        return (None,None,None) # Not supported
+    self.get_cipher = get_cipher
+    get_cipher = get_cipher
+
+    def convert_key(cls, packet, private=False):
+        if types.is_instance(packet, RSAPrivateKey) or types.is_instance(packet, RSAPublicKey) or types.is_instance(packet, DSAPublicKey) or types.is_instance(packet, DSAPrivateKey):
+            if (not private) and (types.is_instance(packet, DSAPrivateKey) or types.is_instance(packet, RSAPrivateKey)):
+                return packet.public_key()
+            else:
+                return packet
+
+        packet = cls._parse_packet(packet)
+        if types.is_instance(packet, OpenPGP.Message):
+            packet = packet[0]
+
+        if packet.key_algorithm_name() == 'DSA':
+            params = dsa.DSAParameterNumbers(
+                      cls._bytes_to_long(packet.key['p']),
+                      cls._bytes_to_long(packet.key['q']),
+                      cls._bytes_to_long(packet.key['g']))
+            public = dsa.DSAPublicNumbers(
+                      cls._bytes_to_long(packet.key['y']),
+                      params)
+            if private:
+                return dsa.DSAPrivateNumbers(cls._bytes_to_long(packet.key['x']), public).private_key(openssl.backend)
+            else:
+                return public.public_key(openssl.backend)
+        else: # RSA
+            public = rsa.RSAPublicNumbers(cls._bytes_to_long(packet.key['e']), cls._bytes_to_long(packet.key['n']))
+            if private:
+                d = cls._bytes_to_long(packet.key['d'])
+                p = cls._bytes_to_long(packet.key['q'])
+                q = cls._bytes_to_long(packet.key['p'])
+                dmp1 = rsa.rsa_crt_dmp1(d, p)
+                dmq1 = rsa.rsa_crt_dmp1(d, q)
+                u = cls._bytes_to_long(packet.key['u'])
+                return rsa.RSAPrivateNumbers(p, q, d, dmp1, dmq1, u, public).private_key(default_backend())
+            else:
+                return public.public_key(default_backend())
+    self.convert_key = convert_key
+    convert_key = convert_key
+
+    def convert_public_key(cls, packet):
+        return cls.convert_key(packet, False)
+    self.convert_public_key = convert_public_key
+    convert_public_key = convert_public_key
+
+    def convert_private_key(cls, packet):
+        return cls.convert_key(packet, True)
+    self.convert_private_key = convert_private_key
+    convert_private_key = convert_private_key
+
+    def _bytes_to_long(cls, b):
+        if hasattr(int, 'from_bytes'):
+            return int.from_bytes(b, byteorder='big', signed=False)
```

Comment:
there is a "long_to_bytes" function and a "bytes_from_long" in Crypto/Util/Number.star that can be used here I think?